### PR TITLE
Add Tls with keystore type config support

### LINKED ISSUE #6640: Broker and Proxy support multiple CA certs
I'm requesting the Broker and Proxy be updated to support multiple CA certs. I have a requirement that internal communication uses an internal CA cert and we allow customers to define their own CA cert alongside.

### PR DESCRIPTION
**Pull request by jiazhai**
Fixes #6640

### Motivation

Add Tls with keystore type config.

### Modifications

Add Tls with keystore type config.

### Verifying this change

- Unit test passed

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by rdhabalia** [comment]
I haven't gone through with the entire change but it seems it touches many different files and I think supporting multiple CA trust stores can be done in an easier way and might not need this big change.
we can always bundle multiple CA trust store in one file and it can be provided to pulsar-broker. We also have similar kind of requirement and we are going to bundle multiple certs into one to support multiple ca usecase.
https://alvinalexander.com/java/java-using-keytool-import-certificate-keystore/


@merlimat can you also please review this PR.

**Comment by jiazhai** [comment]
@rdhabalia Thanks for the comments. The name of the PR seems be a little misleading.  Changed the name to align with the main changes.  It mainly want to support keystore configs. and make multi ca easy config is one result of it.

**Comment by rdhabalia** [comment]
I will try to review it soon

File: pulsar-client-auth-keystoretls/src/test/java/org/apache/pulsar/client/TlsProducerConsumerTestWithAuth.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        log.info("-- Starting {} test --", methodName);
+
+        final int MESSAGE_SIZE = 16 * 1024 + 1;
+        log.info("-- message size --", MESSAGE_SIZE);
+        String topicName = "persistent://my-property/use/my-ns/testTlsClientAuthOverBinaryProtocol"
+                           + System.currentTimeMillis();
+
+        internalSetUpForNamespace();
+
+//        // Test 1 - Using TLS on binary protocol without sending certs - expect failure
```
Comment (sijie) [inline pulsar-client-auth-keystoretls/src/test/java/org/apache/pulsar/client/TlsProducerConsumerTestWithAuth.java:213]: why are these lines commented?

**Comment by sijie** [comment]
@jiazhai I think we need to add those settings to the configuration files and the website documentation. Can you please create an issue to follow up on a change?

File: pulsar-client-auth-keystoretls/src/test/java/org/apache/pulsar/client/TlsProducerConsumerTestWithAuth.java (hunk 1 above)
Comment (jiazhai) [reply pulsar-client-auth-keystoretls/src/test/java/org/apache/pulsar/client/TlsProducerConsumerTestWithAuth.java:213]: Oh, thank, this is for debug reason. will uncomment them

**Comment by jiazhai** [comment]
Hi @rdhabalia Thanks for taking care. If this is too hurry for the review, we could handle it in the following issues and prs.

File: pulsar-client-auth-keystoretls/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationKeyStoreTls.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import org.apache.pulsar.client.api.KeyStoreParams;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.impl.AuthenticationUtil;
+
+/**
+ * This plugin requires these parameters: keyStoreType, keyStorePath, and  keyStorePassword.
+ * This parameter will construct a AuthenticationDataProvider
+ */
+@Slf4j
+public class AuthenticationKeyStoreTls implements Authentication, EncodedAuthenticationParameterSupport {
```
Comment (rdhabalia) [inline pulsar-client-auth-keystoretls/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationKeyStoreTls.java:39]: this is another authentication interface and we can avoid adding a new module for just this interface. can we move this interface under `pulsar-client`

Comment (jiazhai) [reply pulsar-client-auth-keystoretls/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationKeyStoreTls.java:39]: get it.

**Pull request merged**